src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterMetaService.cs

```diff
@@ -58,33 +67,30 @@ public SubjectMeta Get(DataColumnCollection cols, DataRowCollection rows, Subjec
             };
         }
 
-        private static IEnumerable<MetaRow> GetMetaRows(
+        private static List<MetaRow> GetMetaRows(
             List<string> cols,
-            DataRowCollection rows)
+            List<List<string>> metaFileRows)
         {
-            List<MetaRow> metaRows = new List<MetaRow>();
-            foreach (DataRow row in rows)
-            {
-                metaRows.Add(GetMetaRow(cols, row));
-            }
-
-            return metaRows;
+            return metaFileRows.Select(row => GetMetaRow(cols, row)).ToList();
         }
 
-        public static MetaRow GetMetaRow(List<string> cols, DataRow row)
+        public static MetaRow GetMetaRow(List<string> cols, List<string> cells)
         {
-            return CsvUtil.BuildType(CsvUtil.GetRowValues(row), 
-                cols, Enum.GetNames(typeof(MetaColumns)), values => new MetaRow
-            {
-                ColumnName = values[0],
-                ColumnType = (ColumnType) Enum.Parse(typeof(ColumnType), values[1]),
-                Label = values[2],
-                FilterGroupingColumn = values[3],
-                FilterHint = values[4],
-                IndicatorGrouping = values[5],
-                IndicatorUnit = EnumUtil.GetFromString<Unit>(values[6] ?? ""),
-                DecimalPlaces = values[7] == null ? (int?) null : int.Parse(values[7])
-            });
+            return CsvUtil.BuildType(
+                cells, 
+                cols, 
+                Enum.GetNames(typeof(MetaColumns)), 
+                values => new MetaRow
+                {
+                    ColumnName = values[0],
+                    ColumnType = (ColumnType) Enum.Parse(typeof(ColumnType), values[1]),
```

Comment:
I know this isn't your change but this could be `Enum.Parse<ColumnType>(values[1])` without the cast.